lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java

```diff
@@ -1575,11 +1601,51 @@ public ProjectHeader mapRow(ResultSet rs, int rowNum) throws SQLException {
 	public Set<Long> getAllContainerIds(String parentId, int maxNumberIds) throws LimitExceededException{
 		ValidateArgument.required(parentId, "parentId");
 		Long id = KeyFactory.stringToKey(parentId);
-		List<Long> ids = new LinkedList<>();
-		ids.add(id);
-		return getAllContainerIds(ids, maxNumberIds);
+		return getAllContainerIds(Collections.singletonList(id), maxNumberIds);
+	}
+	
+	@Override
+	public SortedMap<Integer, Set<Long>> getAllContainerIdsOrderByDistanceDesc(Collection<Long> parentIds, int minDistance, int limit) {
+		ValidateArgument.required(parentIds, "parentIds");
+		ValidateArgument.requirement(minDistance > 0, "The min distance must be greater than 0");
+		
+		Map<String, Object> parameters = new HashMap<String, Object>(2);
+		
+		parameters.put(PARAM_NAME_IDS, parentIds);
+		parameters.put(PARAM_NAME_MIN_DISTANCE, minDistance);
+		parameters.put(BIND_LIMIT, limit);
+		
+		return namedParameterJdbcTemplate.query(
```

Comment:
if you use the "streamingJdbcTemplate" stream over the results and batch delete as you go.  The "WITH RECURSIVE" would only execute once.

Reply:
Tried with the streaming template and unfortunately one cannot submit other queries on the same connection while streaming :( See https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-implementation-notes.html: "There are some caveats with this approach. You must read all of the rows in the result set (or close it) before you can issue any other queries on the connection, or an exception will be thrown.".